### PR TITLE
Anagrams question and answer 

### PR DESCRIPTION
**Pull request by angelafc001**
Please let me know if there is anything I need to fix or edit.

Thanks!

Angela Ferro Capera

### CODE REVIEW

File: 2020-01-Winter/2_string_manipulation/README.md [hunk 1]
```diff
@@ -116,21 +133,66 @@ Go to [Top](#top)
...
+
+Time Complexity: O(NK). Counting every string in the array is O(N), and counting each character
+in each string is O(K), where K is the max length of a string in the string array.
+
+Space Complexity: O(N)
 
 #### Testing The Solutions OR Driver For Solution
 
-TODO :bug:
+[Available here: ](./2_string_manipulation/ Anagrams/ Anagram.java)
```
Comment (etcadinfinitum) [inline 2020-01-Winter/2_string_manipulation/README.md]: The link text should be identical to the filepath for the .java file. Also, personal nitpick, but spaces in directory or file names are discouraged.

```suggestion
[Available in the repository.](./2_string_manipulation/2. Anagrams/Anagram.java)
```

File: 2020-01-Winter/2_string_manipulation/README.md [hunk 2]
```diff
@@ -116,21 +133,66 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
+### 2. GROUP ANAGRAMS
```
Comment (etcadinfinitum) [inline 2020-01-Winter/2_string_manipulation/README.md]: Provide problem source here, mostly for redundancy
```suggestion

Source: [LeetCode](https://leetcode.com/problems/group-anagrams/)

```

File: 2020-01-Winter/2_string_manipulation/README.md [hunk 3]
```diff
@@ -116,21 +133,66 @@ Go to [Top](#top)
...
+Two strings are anagrams only if the characters counts are the same. 
+
+Another way to solve this problem is by transforming each string into a character count 
+"count", consisting of 26 non-negative integers representing the 
+number of times each character occurs in each string. 
+
+1. Check for an empty array, if the array is not empty, then use a hashtable
+where the key will be represented by a string delimited with '#' characters. For 
+example abbccdddd will be represented as "1#2#2#4" where 1,2, and 4 are the number of 
+times each character occurs in the string.
```
Comment (etcadinfinitum) [inline 2020-01-Winter/2_string_manipulation/README.md]: This wasn't totally clear until I read through the LeetCode problem statement again. Chars which aren't present should be denoted with a 0 placeholder.

File: 2020-01-Winter/2_string_manipulation/README.md [hunk 4]
```diff
@@ -116,21 +133,66 @@ Go to [Top](#top)
...
-TODO :bug:
+Two strings are anagrams only if the characters counts are the same. 
+
+Another way to solve this problem is by transforming each string into a character count 
+"count", consisting of 26 non-negative integers representing the 
+number of times each character occurs in each string. 
+
+1. Check for an empty array, if the array is not empty, then use a hashtable
+where the key will be represented by a string delimited with '#' characters. For 
+example abbccdddd will be represented as "1#2#2#4" where 1,2, and 4 are the number of
```
Comment (etcadinfinitum) [inline 2020-01-Winter/2_string_manipulation/README.md]:
```suggestion
example abbccdddd will be represented as "1#2#2#4#0...#0" where 1,2, and 4 are the number of 
```

File: 2020-01-Winter/2_string_manipulation/README.md [hunk 5]
```diff
@@ -116,21 +133,66 @@ Go to [Top](#top)
...
+
+5. Add the original (unsorted) string as a new value of the key.
+ 
+6. Return the list of list of anagrams
+
+Time Complexity: The outer loop has complexity O(N) as we iterate through each string. 
+Then, we sort each string in O(N log N) time. Therefore, the time complexity of this 
+algorithm is O(N Log N)
+
+Space Complexity: O(N)
```
Comment (etcadinfinitum) [inline 2020-01-Winter/2_string_manipulation/README.md]: Missing the solution code here; add it in like so:

````suggestion

```java
public static List<List<String>> groupAnagramsOpt(String[] strs) {
    if (strs.length == 0) {
        return new ArrayList<>();
    }
    Map<String, List<String>> map = new HashMap<>();    
    for (String st:strs) {
        char[] chars = st.toCharArray();    
        Arrays.sort(chars);    
        String key = String.valueOf(chars); // Converting the sorted char sequence into string    
        if (!map.containsKey(key)) {
            map.put(key, new ArrayList<String>());
        }    
        map.get(key).add(st);
    }
    return new ArrayList(map.values());
}
```

````

**Comment by angelafc001** [comment]
Hey Lizzy, I have made your changes.

**Pull request closed without merging**